gammapy/cube/psf_kernel.py

```diff
@@ -0,0 +1,310 @@
+            geom = WcsGeom.create(skydir=center, binsz=binsz, npix=npix, proj=geom.projection,
+                                  coordsys=geom.coordsys, axes=geom.axes)
+        return cls(table_psf_to_kernel_map(table_psf, geom, normalize, factor))
+
+    @classmethod
+    def from_gauss(cls, geom, sigma, max_radius=None, containment_fraction=0.99,
+                   normalize=True, factor=4):
+        """Create Gaussian PSF.
+
+        This is used for testing and examples.
+        The map geometry parameters (pixel size, energy bins) are taken from `geom`.
+        The Gaussian width ``sigma`` is a scalar,
+
+        TODO : support array input if it should vary along the energy axis.
+
+        Parameters
+        ----------
+        geom : `~gammapy.map.WcsGeom`
+            Map geometry
+        sigma : `~astropy.coordinates.Angle` or float
+            Gaussian width. Assume degrees if float input
+        max_radius : `~astropy.coordinates.Angle` or float
+            Desired kernel map size. Implicit unit is degree.
+        normalize : bool
+            normalize the PSF kernel (per energy)
+        factor : int
+            the oversample factor to compute the PSF
+
+        Returns
+        -------
+        kernel : `~gammapy.cube.PSFKernel`
+            the kernel Map with reduced geometry according to the max_radius
+        """
+        sigma = Angle(sigma, 'deg')
+
+        if max_radius is None:
+            max_radius = Gauss2DPDF(sigma.to('deg').value).containment_radius(
+                containment_fraction=containment_fraction)
+
+        max_radius = Angle(max_radius, 'deg')
+
+        # Create a new geom according to given input
+        center = geom.center_coord[:2]
+        binsz = Quantity(np.abs(geom.wcs.wcs.cdelt[0]), 'deg')
+        npix = 2 * int(max_radius / binsz) + 1
+        geom = WcsGeom.create(skydir=center, binsz=binsz, npix=npix, proj=geom.projection,
+                              coordsys=geom.coordsys, axes=geom.axes)
+
+        rad = Angle(np.linspace(0., max_radius, 200), 'deg')
+
+        table_psf = TablePSF.from_shape(shape='gauss', width=sigma, rad=rad)
+
+        return cls(table_psf_to_kernel_map(table_psf, geom))
+
+    def to_map(self):
+        return self._psf_kernel_map
+
+    def write(self, *args, **kwargs):
+        psf_kernel_map = self.to_map()
+        psf_kernel_map.write(*args, **kwargs)
+
+    def apply(self, map):
+        """Apply the kernel to an input Map.
+
+        Parameters
+        ----------
+        map : `~gammapy.maps.WcsNDMap`
+            the model map to be convolved with the PSFKernel.
+            It should have the same MapGeom than the current PSFKernel.
+
+        Returns
+        -------
+        convolved_map : `~gammapy.maps.Map`
+            the convolved map
+        """
+        # TODO : check that the MapGeom are consistent
+        convolved_map = Map.from_geom(geom=map.geom, unit=map.unit, meta=map.meta)
```

Comment:
I'm not sure how `kernel.apply(map)` should work.
This might be the bottleneck in the likelihood eval for a lot of analyses with Gammapy.
So what we have here, to first make a new Map with zero-filled data, and then to replace those values with convolved other values is not efficient.

One solution could be to do an in-place apply (just delete the line making the `convolved_map` here, and below use `map.data` instead of `data`. Suggest to make that change now.

If someone wants to keep an unmodified copy they would do this:
```
map_convolved = kernel.apply(map.copy())
```
(copy coming shortly)

We could also move this elsewhere, e.g. `map.apply_psf_kernel(psf_kernel)` or to the model evaluator.
Suggest to defer this decision of code organisation to later, when we have a second map kernel or apply method we support.

Reply:
OK for the copy. Good point. Yet, as far as I understand astropy convolve_fft and scipy fftconvolve do not perform in-place convolution so the gain in term of memory allocation is likely small no?

We could have a general convolution function or method on WcsNDMaps indeed. It could check for the consistency of geoms.